### ISSUE TITLE
Bug: TypeError: Cannot read property 'length' of undefined

### ISSUE THREAD

**Issue by iFarmGolems** [buehler/typescript-hero#403]
#### System information

- OS: Windows 10
- TypeScript Hero version: 2.2.0
- VSCode version: 1.20.0

#### Description

I'm seeing this error when TSH Resolver kicks in:

2018-02-09T15:03:47.206Z - error: [DeclarationManager] A workspace did encounter an error.
2018-02-09T15:03:47.207Z - error: [WorkspaceDeclarations] Error during indexing of workspacefiles. {"error":"TypeError: Cannot read property 'length' of undefined","workspace":"c:\\Users\\patrikj\\DEV\\sandbox"}

##### Steps to reproduce

1. Open Output tab in VS Code
2. Choose TypeScript Hero in selection
3. Open some .ts file

**Comment by buehler** [comment]
I experienced this one as well.. but not always :-/

**Comment by El-Tee-E** [comment]
Same issue here, same repro steps.

On Ubuntu 16
With latest VSCode (Version 1.20.0)

Commit c63189deaa8e620f650cc28792b8f5f3363f2c5b
Date 2018-02-07T17:10:15.949Z
Shell 1.7.9
Renderer 58.0.3029.110
Node 7.9.0
Architecture x64 

In angular 4.1 project (project is quite large)

And the 'fire icon' keeps 'burning'.

**Comment by Cronkan** [comment]
Having the same problem, removing node_modules "solves the problem". 
Reinstalling with npm install the problem occurs again.
debuginfo:
```
debug: [WorkspaceDeclarations] Calculated node module search. {"globs":["node_modules/@angular/animations/**/*.d.ts","node_modules/@angular/common/**/*.d.ts","node_modules/@angular/compiler/**/*.d.ts","node_modules/@angular/core/**/*.d.ts","node_modules/@angular/forms/**/*.d.ts","node_modules/@angular/http/**/*.d.ts","node_modules/@angular/platform-browser/**/*.d.ts","node_modules/@angular/platform-browser-dynamic/**/*.d.ts","node_modules/@angular/router/**/*.d.ts","node_modules/bootstrap/**/*.d.ts","node_modules/core-js/**/*.d.ts","node_modules/mo-js/**/*.d.ts","node_modules/rxjs/**/*.d.ts","node_modules/zone.js/**/*.d.ts","node_modules/@angular-devkit/core/**/*.d.ts","node_modules/@angular/cli/**/*.d.ts","node_modules/@angular/compiler-cli/**/*.d.ts","node_modules/@angular/language-service/**/*.d.ts","node_modules/@types/jasmine/**/*.d.ts","node_modules/@types/jasminewd2/**/*.d.ts","node_modules/@types/node/**/*.d.ts","node_modules/codelyzer/**/*.d.ts","node_modules/jasmine-core/**/*.d.ts","node_modules/jasmine-spec-reporter/**/*.d.ts","node_modules/karma/**/*.d.ts","node_modules/karma-chrome-launcher/**/*.d.ts","node_modules/karma-cli/**/*.d.ts","node_modules/karma-coverage-istanbul-reporter/**/*.d.ts","node_modules/karma-jasmine/**/*.d.ts","node_modules/karma-jasmine-html-reporter/**/*.d.ts","node_modules/protractor/**/*.d.ts","node_modules/ts-node/**/*.d.ts","node_modules/tslint/**/*.d.ts","node_modules/typescript/**/*.d.ts"],"ignores":["node_modules/@angular/animations/**/node_modules/**/*","node_modules/@angular/common/**/node_modules/**/*","node_modules/@angular/compiler/**/node_modules/**/*","node_modules/@angular/core/**/node_modules/**/*","node_modules/@angular/forms/**/node_modules/**/*","node_modules/@angular/http/**/node_modules/**/*","node_modules/@angular/platform-browser/**/node_modules/**/*","node_modules/@angular/platform-browser-dynamic/**/node_modules/**/*","node_modules/@angular/router/**/node_modules/**/*","node_modules/bootstrap/**/node_modules/**/*","node_modules/core-js/**/node_modules/**/*","node_modules/mo-js/**/node_modules/**/*","node_modules/rxjs/**/node_modules/**/*","node_modules/zone.js/**/node_modules/**/*","node_modules/@angular-devkit/core/**/node_modules/**/*","node_modules/@angular/cli/**/node_modules/**/*","node_modules/@angular/compiler-cli/**/node_modules/**/*","node_modules/@angular/language-service/**/node_modules/**/*","node_modules/@types/jasmine/**/node_modules/**/*","node_modules/@types/jasminewd2/**/node_modules/**/*","node_modules/@types/node/**/node_modules/**/*","node_modules/codelyzer/**/node_modules/**/*","node_modules/jasmine-core/**/node_modules/**/*","node_modules/jasmine-spec-reporter/**/node_modules/**/*","node_modules/karma/**/node_modules/**/*","node_modules/karma-chrome-launcher/**/node_modules/**/*","node_modules/karma-cli/**/node_modules/**/*","node_modules/karma-coverage-istanbul-reporter/**/node_modules/**/*","node_modules/karma-jasmine/**/node_modules/**/*","node_modules/karma-jasmine-html-reporter/**/node_modules/**/*","node_modules/protractor/**/node_modules/**/*","node_modules/ts-node/**/node_modules/**/*","node_modules/tslint/**/node_modules/**/*","node_modules/typescript/**/node_modules/**/*"],"workspace":"c:\\[REDACTED]"}
console.ts:123 [Extension Host] 2018-02-18T12:09:12.062Z - info: [WorkspaceDeclarations] Found 2531 files in workspace. {"workspace":"c:\\[REDACTED]"}
```
Package.json
```
{
  "name": "[REDACTED]",
  "version": "0.0.0",
  "license": "MIT",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build --prod",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "^5.0.0",
    "@angular/common": "^5.0.0",
    "@angular/compiler": "^5.0.0",
    "@angular/core": "^5.0.0",
    "@angular/forms": "^5.0.0",
    "@angular/http": "^5.0.0",
    "@angular/platform-browser": "^5.0.0",
    "@angular/platform-browser-dynamic": "^5.0.0",
    "@angular/router": "^5.0.0",
    "bootstrap": "^4.0.0",
    "core-js": "^2.4.1",
    "mo-js": "^0.288.2",
    "rxjs": "^5.5.2",
    "zone.js": "^0.8.14"
  },
  "devDependencies": {
    "@angular-devkit/core": "^0.3.1",
    "@angular/cli": "1.6.3",
    "@angular/compiler-cli": "^5.0.0",
    "@angular/language-service": "^5.0.0",
    "@types/jasmine": "~2.5.53",
    "@types/jasminewd2": "~2.0.2",
    "@types/node": "~6.0.60",
    "codelyzer": "^4.0.1",
    "jasmine-core": "~2.6.2",
    "jasmine-spec-reporter": "~4.1.0",
    "karma": "~1.7.0",
    "karma-chrome-launcher": "~2.1.1",
    "karma-cli": "~1.0.1",
    "karma-coverage-istanbul-reporter": "^1.2.1",
    "karma-jasmine": "~1.1.0",
    "karma-jasmine-html-reporter": "^0.2.2",
    "protractor": "~5.1.2",
    "ts-node": "~3.2.0",
    "tslint": "~5.7.0",
    "typescript": "~2.4.2"
  }
}
```

**Comment by dubcdr** [comment]
I'm getting this problem as well. I think it has something to do with angular cli.

**Comment by buehler** [comment]
Already hat that issue one time.. This is a compiler bug: 
https://github.com/Microsoft/TypeScript/issues/20792 &
https://github.com/Microsoft/TypeScript/issues/21801

Which will be fixed, as soon as 2.8 is out.
